packages/orchestration/src/workflow/workflow-manager.ts

```diff
@@ -75,11 +76,18 @@ export class WorkflowManager {
     requiredModules?: Set<string>,
     optionalModules?: Set<string>
   ) {
+    const finalFlow = flow instanceof OrchestratorBuilder ? flow.build() : flow
+
     if (WorkflowManager.workflows.has(workflowId)) {
-      throw new Error(`Workflow with id "${workflowId}" is already defined.`)
-    }
+      const areStepsEqual = deepEqualObj(
```

Comment:
isn't this enough?
`JSON.stringify(finalFlow) === JSON.stringify(WorkflowManager.workflows.get(workflowId)!.flow_)`

Reply:
do you think it is better? couldn't we have side effects while moving forward?

Reply:
I don't think so. In case this evolve to contain any circular reference of an object, we'll need to use that utils we have `stringifyCircular`